### DISCUSSION TITLE
Attribute forms, text editing and cursors

### DISCUSSION THREAD

**Discussion by Demexyz** [opengisch/QField#3991] [Q&A]
Hi! I wanted to start a discussion about how attribute forms and text editing/input works in QField, and maybe even get some help with issues I've been encountering.

I've been trying out QField for my fieldwork for the past week or so and overall I'm very happy with how its working. But when filling out attribute forms I've been running into a variety of inconvenient/odd behaviour. I'm not sure what is determined by OS/keyboard and what might be QFields (intended?) behaviour, but I thought it might be useful to hear other peoples experiences. For reference I'm using a Samsung Galaxy Tab Active3, mostly with the native Samsung keyboard.

The first issue I noticed was when inputting into a field real numbers field. When typing, Android would correctly show a numpad-style keypad. But when inserting a number with a decimal point the input would be "autocorrected" like text, moving the point to the end of the number (e.g. "3.2" > "32."). From what I've been able to find out, this seems to primarily be an issue with the Samsung keyboard.

The second thing I've been trying to figure out is how to work with the multiline text edit widget. Here the main oddity seems to be that tapping anywhere in the textbox a _second_ time is interpreted as exiting the text edit mode (though the text cursor is still visible). Doing this also makes the keyboard flash back and forth, moving the whole UI along with it. This means that text in the textbox can not be selected by double- or triple tapping. Nor can the text cursor easily be moved to a different position, as this will close the keyboard. So far I've had to work around this by using the keyboard cursor keys.

A third oddity I've noticed is that when selecting a field set to use a value relation and completer, the keyboard language will change (from Swedish, which in this case is the primarily used language, to English).

Finally, I've noticed for value map/combo box that the last, blank row where an additional value/description can be added to the table seems to show up as an option in drop-down list (whereas they do not in QGIS). This is a minor thing, but am I missing something?

Sorry for the wall of text, hoping somebody finds this interesting or has feedback/answers!

**Comment by 9ls1** [comment]
Thanks for sharing your thoughts and experiences.
1st. issue: I have noticed the same behavior om my Samsung Galaxy Tab S4. A "workaround/solution" seems to be to wait for a moment before entering the decimal digit(s).
2nd. issue: Haven't been using multi-line text in QField so far.
3rd. issue: I'm always using American/English - both in QGIS and QField. However, my attribute names and items in lists may be in Norwegian.
4th. issue: I have noticed the same.
